### PR TITLE
Grunt Build Process

### PR DESCRIPTION
**Pull request by pixelkritzel**
With this pull request you can use grunt for (auto-)compiling the src/js-files

Commands:

* `grunt build` for a one time build
* `grunt watch-src` for a build and rebuilding of single files after file change

To use this please run `npm install` in the project directory

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+// Because you can't specify a directory as target for grunt-browserify we have to generate the file configuration at runtime
```
Comment (skddc) [inline Gruntfile.js]: Sorry about all the Hound comments earlier. I removed them. This line is indeed a bit long, though. ;)

File: Gruntfile.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  grunt.loadNpmTasks('grunt-contrib-watch');
+  grunt.loadNpmTasks('grunt-contrib-clean');
+  grunt.loadNpmTasks('grunt-browserify');
+
+  grunt.registerTask('build', [
+    'clean',
+    'browserify'
+  ]);
+
+  grunt.registerTask('watch-src', [
```
Comment (skddc) [inline Gruntfile.js]: Could we just call it "watch"? Or is there a specific reason, like reserved task names or something?

**Comment by skddc** [comment]
Cool, thanks!

**Comment by skddc** [comment]
Just tested it and works like a charm. The console output is a bit weird, but I think that's not too important for now.

See inline code comments for 2 other tiny things.

File: Gruntfile.js (hunk 2 above)
Comment (pixelkritzel) [inline Gruntfile.js]: If you call it 'watch' it starts the watch task again. It is a inifinite loop.

**Comment by pixelkritzel** [comment]
The output is from the 'grunt-browserify' plugin. There is not much I could change in reasonable time.

**Comment by skddc** [comment]
Argh, turning off Hound. :/

File: Gruntfile.js (hunk 2 above)
Comment (skddc) [inline Gruntfile.js]: Ah, got it. Thanks.

**Comment by skddc** [comment]
Merging this without +1s because it's being merged to a branch. I'm opening a PR to master when the whole ES6 switch is ready.

**Comment by skddc** [comment]
Thanks a lot, @pixelkritzel! And welcome to remoteStorage! :D

cheers-2

**Pull request merged**